### PR TITLE
[RFR] Add marble toboard

### PR DESCRIPTION
**Pull request by xavierhans**
Allow the player 1 to add a marble to the board [Trello](https://trello.com/c/O3GhGgEJ/3-04-as-marcel-given-its-my-turn-to-play-i-want-to-tap-on-an-available-spot-to-add-a-marble)

- [x] BoardCase component
- [x] Add marble when a click is detected
- [ ] Test

### CODE REVIEW

File: game/Model.js [hunk 1]
```diff
@@ -15,5 +15,6 @@ export const addMarbleToBoard = (board, x, y, player) => {
     });
 
     newBoard[x][y] = player;
+    console.log(newBoard)
```
Comment (alexisjanvier) [inline game/Model.js]: debug spoted ?

File: game/BoardCase.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                source={require('./img/red_piece.png')}
+                resizeMode="contain"
+            />
+        );
+    };
+
+    render() {
+        return (
+            <TouchableHighlight
+                onPress={() => this.props.clickCase(this.props.x, this.props.y)}
```
Comment (alexisjanvier) [inline game/BoardCase.js]: where does this `this.props.x` and `this.props.y` come from? Is it React Native specific ?

And second question : there is no `PropTypes` in React Native ?

File: game/Model.js [hunk 3]
```diff
@@ -15,5 +15,6 @@ export const addMarbleToBoard = (board, x, y, player) => {
     });
```
Comment (alexisjanvier) [inline game/Model.js]: `var newBoard = board.map(function(arr) {` => `const  newBoard = board.map(arr => { `

File: game/BoardCase.js (hunk 2 above)
Comment (xavierhans) [reply game/BoardCase.js]: X and Y are the position of the case on the board. It's not React Native specific.
I forgot the propTypes for this one.

Comment (alexisjanvier) [reply game/BoardCase.js]: So if `clickCase`, `x` and `y` come from props, can't you just send a `clickCase` function that already contains these variables? (unless you need it elsewhere!).

Also, I find it more readable to extract props at the beginning of the render : 
```
const { clickCase, x, y } = this.props
```

File: game/BoardCase.test.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import BoardCase from './BoardCase';
+
+import renderer from 'react-test-renderer';
+
+it('renders without crashing', () => {
```
Comment (alexisjanvier) [inline game/BoardCase.test.js]: This test is not very significant :)

File: game/Screen.js [hunk 5]
```diff
@@ -1,26 +1,38 @@
...
+            board: addMarbleToBoard(this.state.board, x, y, this.state.player),
+        });
+    };
+
     render() {
-        let board = createEmptyBoard();
         return (
             <View style={styles.board}>
-                {board.map((row, rowKey) => {
+                {this.state.board.map((row, rowKey) => {
```
Comment (alexisjanvier) [inline game/Screen.js]: You can avoid braket and return 
```
{this.state.board.map((row, rowKey) => (
    <View style={styles.row} key={rowKey}>
        ....
```

File: game/Screen.js [hunk 6]
```diff
@@ -1,26 +1,38 @@
...
                         <View style={styles.row} key={rowKey}>
                             {row.map((elem, elemKey) => {
                                 return (
-                                    <Image
-                                        key={elemKey}
-                                        source={require('./img/white_piece.png')}
-                                        style={styles.case}
-                                        resizeMode="contain"
+                                    <BoardCase
+                                        key={rowKey + '-' + elemKey}
```
Comment (alexisjanvier) [inline game/Screen.js]: Prefer template string : 
```
`${rowKey}-${elemKey}`
```

File: game/Screen.test.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import GameScreen from './Screen';
+
+import renderer from 'react-test-renderer';
+
+it('renders without crashing', () => {
+    const navigation = {
+        navigate: jest.fn(),
+    };
+    expect(
```
Comment (alexisjanvier) [inline game/Screen.test.js]: As you used a mock, you can test if it was called.

**Pull request merged**